var/spack/repos/builtin/packages/elpa/package.py

```diff
@@ -137,6 +137,20 @@ def headers(self):
     build_directory = "spack-build"
     parallel = False
 
+    def flag_handler(self, name, flags):
+        if self.spec.satisfies("+rocm"):
+            if name == "ldlibs":
+                # NOTE: hipcc may be used to build some files but the final link
+                # step is not done using hipcc. We must provide the libs.
+                # SPACK_LINK_DIRS already contains the library path.
+                flags.extend(
+                    [
+                        "-l{}".format(library.split("lib")[1])
+                        for library in self.spec["hip"].package.libraries
+                    ]
+                )
```

Comment:
```suggestion
                flags.extend(self.spec["hip"].libs.link_flags.split())
```
The `libs` property gets all libraries associated with the package. The `libraries` attribute is used for external detection and is not intended to be a complete list, merely sufficient to identify the package.